File: rest-client/src/main/java/org/acme/restclient/Country.java
package org.acme.restclient;

import java.util.List;

public class Country {

    public String name;
    public String alpha2Code;
    public String capital;
    public List<Currency> currencies;

    public static class Currency {
        public String code;
        public String name;
        public String symbol;
    }

}

File: rest-client-kotlin-quickstart/src/main/kotlin/org/acme/restclient/Country.kt
package org.acme.restclient

class Country {

    lateinit var name: String
    lateinit var alpha2Code: String
    lateinit var capital: String
    lateinit var currencies: List<Currency>

    class Currency {
        lateinit var code: String
        lateinit var name: String
        lateinit var symbol: String
    }

}
